src/providers/bitcoin/BitcoinSwapProvider.js

```diff
@@ -145,24 +153,15 @@ export default class BitcoinSwapProvider extends Provider {
         block = await this.getMethod('getBlockByNumber')(blockNumber, true)
       } catch (e) { }
       if (block) {
-        const transactions = block.transactions
-        const txs = await Promise.all(transactions.map(async transaction => {
-          return this.getMethod('decodeRawTransaction')(transaction)
-        }))
-        initiateSwapTransaction = txs
-          .map(transaction => transaction._raw.data)
-          .map(transaction => transaction.vout
-            .map(vout => { return { txid: transaction.txid, scriptPubKey: vout.scriptPubKey, value: vout.valueSat } }))
-          .reduce((acc, val) => acc.concat(val), [])
-          .find(txInfo => txInfo.scriptPubKey.hex === sendScript && txInfo.value === value)
+        initiateSwapTransaction = block.transactions.find(tx =>
+          this.doesTransactionMatchSwapParams(tx, value, recipientAddress, refundAddress, secretHash, expiration)
+        )
         blockNumber++
       }
       await sleep(5000)
```

Comment:
Just noticed `sleep(5000)`

This means a 5 second delay correct? Should we reduce this to enable faster block scanning?

Reply:
Yes 5s. I'm not sure what to exactly do with the wait times because it depends on quite a few factors. The minimum blocktime on ethereum is for example 15s atm. I just picked 5s randomly 🤷‍♂️ 
I think eventually we need to think of this properly, like events on blocks being available, perhaps an option on the provider for how often to scan the chain etc.

Reply:
Agreed @monokh  I think we also need to think about how fast we want to scan the chain in certain instances. For example if we're significantly behind, then it would make sense to scan asynchronously as fast as possible, but if the current block is latest block then a longer delay makes sense.

Reply:
Maybe it would makes sense to check if it's the latest block, and if so, add `sleep` in there

Reply:
Wouldn't you then be spamming the node? 😕  feel a little uncomfortable about something like that atm

Reply:
Yep we'd have to check how much this will affect the performance of the node.